File: 1 Two Sum/1 Two Sum.swift
class Solution {
  func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
    var numsDict: [Int: Int] = [:]

    for i in 0..<nums.count {
      if let found = numsDict[target - nums[i]] {
        return [found, i]
      }

      numsDict[nums[i]] = i
    }

    return []
  }
}

File: 1 Two Sum/1 Two Sum.ts
function twoSum(nums: number[], target: number): number[] {
  const numsMap: Map<number, number> = new Map();

  for (let i = 0; i < nums.length; i++) {
    const remaining: number = target - nums[i];

    if (numsMap.has(remaining)) {
      return [numsMap.get(remaining)!, i];
    }

    numsMap.set(nums[i], i);
  }

  return [];
}
